sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -81,6 +89,141 @@ def __init__(
             )
         self.job_kind = job_kind
 
+    def train(
+        self,
+        name: str = None,
+        namespace: str = None,
+        num_workers: int = 1,
+        num_procs_per_worker: int = 1,
+        storage_config: Dict[Literal["size", "storage_class"], str] = None,
+        model_provider_parameters: HuggingFaceModelParams = None,
+        dataset_provider_parameters: S3DatasetParams = None,
+        train_parameters: HuggingFaceTrainParams = None,
+        resources_per_worker: Dict[Literal["gpu", "cpu", "memory"], any] = None,
+    ):
+        """
+        Higher level train api
+        """
+        if (
+            not name
+            or not namespace
+            or not storage_config
+            or not model_provider_parameters
+            or not dataset_provider_parameters
+            or not train_parameters
+            or not resources_per_worker
+        ):
+            raise ValueError("One of the required parameters is None")
```

Comment:
Is this correct ?
For namespace, can we do the similar assign as here: https://github.com/kubeflow/training-operator/blob/master/sdk/python/kubeflow/training/api/training_client.py#L151
For other parameters, can we verify only required parameters that need to be set (e.g. `name`)

Reply:
Except namespace all else are required.

Reply:
@deepanker13 Please can you remove namespace for this check and make the same assignment as for other APIs ?
e.g.
```
namespace = namespace or self.namespace
```

Reply:
Also, why `storage_config` is required ?
We discussed with @johnugeorge that we can set some default values for Storage Config (10 Gi and default storage class) if that is not set.
E.g.
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: storage-initializer
spec:
  accessModes:
  - ReadOnlyMany
  resources:
    requests:
      storage: 10Gi
```